src/ringct/rctSigs.cpp

```diff
@@ -1748,4 +1749,52 @@ namespace rct {
       key mask;
       return decodeRctSimple(rv, sk, i, mask, hwdev);
     }
+
+    key getCommitment(const cryptonote::transaction &tx, const std::size_t output_idx) {
```

Comment:
Personally, I would move this to the crypotonote format utils. Right now the `cryptonote_basic` folder depends on `ringct`, not the other way around.

Reply:
Hmm, think you mixed the two up here. `cryptonote_basic` doesn't depend on `ringct`, and `ringct` does depend on `cryptonote_basic`. The reason I placed it here initially is because `is_coinbase` requires the dep to `cryptonote_basic`, and `zeroCommitVartime` requires the ringct dep.

Reply:
This PR to master should make things a bit clearer where the dependencies actually are: https://github.com/monero-project/monero/pull/9919. Because, yes, `libringct` actually linked against `libcryptonote_basic` (for a single convenience function), there was a larger hidden dependency where `libcryptonote_basic` needed to link against `libringct_basic`, but included a header from `libringct`. And `libcryptonote_basic` will always include headers from `libringct_basic`: `rctTypes.h`, since `cryptonote::transaction` has a field member of a type defined in `libringct_basic`.